servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/DefaultNettyConnection.java

```diff
@@ -140,6 +139,28 @@
     private volatile DataObserver dataObserver;
     private final boolean isClient;
     private final UnaryOperator<Throwable> enrichProtocolError;
+    private final TerminalSignalConsumer cleanupStateConsumer = new TerminalSignalConsumer() {
```

Comment:
I debated having `DefaultNettyConnection implement TerminalSignalConsumer` but was hesitant to expose  this on public api